src/ctrace/src/ctrace_action.erl

```diff
@@ -0,0 +1,60 @@
+% Licensed under the Apache License, Version 2.0 (the "License"); you may not
+% use this file except in compliance with the License. You may obtain a copy of
+% the License at
+%
+%   http://www.apache.org/licenses/LICENSE-2.0
+%
+% Unless required by applicable law or agreed to in writing, software
+% distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+% WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+% License for the specific language governing permissions and limitations under
+% the License.
+
+-module(ctrace_action).
+
+-export([
+    sample/1,
+    report/1,
+    report_longer_than/1
+]).
+
+-type action_fun()
+    :: fun((Span :: passage_span:span()) -> boolean()).
+
+-spec sample(
+        [SamplingRate :: float() | integer()]
+    ) -> action_fun().
+
+sample([SamplingRate]) ->
+    fun(_Span) -> rand:uniform() < SamplingRate end.
+
+-spec report(
+        [ReporterId :: passage:tracer_id()]
+    ) -> action_fun().
+
+report([_TracerId]) ->
+    fun(_Span, Options) ->
+        passage_pd:finish_span(Options),
+        true
+    end.
+
+report_longer_than([_TracerId, Threshold]) ->
```

Comment:
I'll remove this. Since as @davisp pointed out [here](https://github.com/apache/couchdb/pull/2286#issuecomment-551164613) it is not really useful.

Reply:
removed in https://github.com/apache/couchdb/pull/2309/commits/50e7a8172719a4c0002a2e9afa73b5926bfe3c46